src/contributing/release_process.md

```diff
@@ -0,0 +1,164 @@
+numbering schemes, which are not necessarily aligned with each other. However, since the primary
+component of the Parsec project is the Parsec service, it is the version number of the Parsec
+service that is used to denote the version number of each Parsec release.
+
+For example, the Parsec `0.7.0` release, which was made in March 2021, was so called because the
+Parsec service was at version `0.7.0` for that particular release, even though the Parsec tool was
+at `0.3.0`, and the various other component repositories were tagged according to their own schemes.
+
+## Common Rust Dependencies
+
+All of the Rust crates that make up a Parsec release will be published at a single common version
+number for the release. This is to ensure that the Parsec service and Parsec tool will always be
+consuming a single common version of each crate. Having multiple versions of the same crate in a
+single release makes it difficult for packaging the release into Operating System distributions, and
+so this is avoided.
+
+For example, the Parsec `0.7.0` release was composed from version `0.7.0` of the Parsec service, and
+`0.3.0` of the Parsec tool. The
+[parsec-interface-rs](https://github.com/parallaxsecond/parsec-interface-rs) crate is consumed into
+both of these components. Its version was set at `0.24.0`, and this single common version was
+consumed into both the Parsec service and the Parsec tool for the combined `0.7.0` release. A
+similar strategy was adopted for other common Rust crates in the project.
+
+For more information about the Rust crate dependencies, including full details on how to release
+them individually, please refer to the [Package Management and Versioning
+Guide](package_management.md).
+
+## Release Codenames
+
+The Parsec project does not employ codenames for releases. Future releases are simply named
+according to the expected version number of the Parsec service.
+
+## Code Freeze and Release Branching
+
+Shortly before each Parsec release, a code freeze will be put into effect, and a release branch will
+be cut from main. Precise details around code freezing and branching will be communicated through
+the community channels by the maintainers on each release cycle. Branching details may vary, and not
+all repositories will necessarily be branched for every release.
+
+Code freeze dates may also vary depending on the content of each release. Typically, a code freeze
+will occur at around 2-4 weeks before the target release date.
+
+During the code freeze period, code submissions will only be accepted if they address critical
+defects or other issues that are blocking the release.
+
+Shortly after the code freeze date, a release candidate will be published for each of the Parsec
+service and Parsec tool. The availability of this release candidate will be communicated to
+downstream consumers such as Operating System maintainers, giving them the opportunity to perform
+integration testing before the release is finalized.
```

Comment:
Just to be clear, I think it's fine for the release candidate to have a "real" semantic version, like `0.8.0`. If any bugs are discovered during the code freeze period, patch versions can be released subsequently (`0.8.1`, `0.8.2`, etc..). Hopefully, it won't be needed to add a new feature during the code feeze period.

This is to not have to use `-rc*` tags which can be messy with Cargo use.

Reply:
This is the kind of detail I'm intentionally glossing over for now, because I would like to see how the "practice run" for this process works out later in the year. Once we've done that first try, I think we'll probably come back to this page and add quite a few specific details and refinements.